Update SessionTestsMixin from Django project

jdufresne opened the pull request:
Helps ensure django-redis used as a session complies with Django expectations. Includes updated tests and new tests. Allows closely following the upstream session compliance test.

Copied from Django's sessions test suite: https://github.com/django/django/blob/master/tests/sessions_tests/tests.py

niwinz commented:
Sorry for the so late review and merge! this month i'm so busy  (traveling...) can you please fix the conflicts? I have merged your other PRs and they are in conflict with this one and the other one...
I'm currently on my holidays, so i will bump a version in two weeks (after 5 january...)

jdufresne commented:
It is no problem at all. Thanks so much for maintaining a great and useful package.

I have rebased all PRs. The CI failures are due to an upstream lz4 issue that I've documented and fixed here:  https://github.com/python-lz4/python-lz4/issues/75. Waiting on a new release before these tests will be green.
